tremor-pipeline/src/op/qos/wal.rs

```diff
@@ -193,17 +167,13 @@ pub struct Wal {
 }
 
 op!(WalFactory(_uid, node) {
-    if let Some(map) = &node.config {
-        let config: Config = Config::new(map)?;
-
-        if let (None, None) = (config.max_elements, config.max_bytes) {
-            Err(ErrorKind::BadOpConfig("WAL operator needs at least one of `max_elements` or `max_bytes` config entries.".to_owned()).into())
-        } else {
-            Ok(Box::new(Wal::new(node.id.to_string(), config)?))
-        }
+    let map = node.config.as_ref().ok_or_else(|| ErrorKind::MissingOpConfig(node.id.to_string()))?;
+    let config: Config = Config::new(&map)?;
 
+    if config.max_elements.or(config.max_bytes).is_none() {
+        Err(ErrorKind::BadOpConfig("WAL operator needs at least one of `max_elements` or `max_bytes` config entries.".to_string()).into())
```

Comment:
just as a side note: we should delete `max_bytes` here. In another PR, until we find out how to fix this in sled or have another storage engine for the wal.